openedx/core/djangoapps/programs/models.py

```diff
@@ -65,6 +65,15 @@ class ProgramsApiConfig(ConfigurationModel):
         default=False
     )
 
+    max_retries = models.PositiveIntegerField(
+        verbose_name=_("Maximum API Retries"),
+        default=3,
+        help_text=_(
+            "When making requests to the Programs API, make at most this many attempts "
```

Comment:
Naming nitpicks: the task used to award certificates makes requests to internal, Programs, and Credentials APIs. This might be more accurate left simply as:

> Make at most this many attempts to award programs certificates.

The field name could be more precisely named `max_certification_retries` (corresponding to the `enable_certification` field), and the verbose name might also be more accurate as "Maximum Certification Retries."